### PR TITLE
86 genlayer simulator backend fixes

### PR DESCRIPTION
**Pull request by sevenearths**
Only one remaining is:

- The Contract state is mixing the input parameters and the self. should be updated an returned in the get_contract_state call

Please read [this comment](https://github.com/yeagerai/genlayer-simulator/issues/86#issuecomment-2083855120)

### CODE REVIEW

**Comment by sevenearths** [comment]
> @sevenearths please update with main and adapt to the breaking changes from #92

Done!

File: rpc/server.py [hunk 1]
```diff
@@ -400,11 +402,11 @@ async def call_contract_function(
...
     )
 
-    return {
-        "status": "success",
-        "message": f"Function '{function_name}' called on contract at {contract_address} with args {args}.",
-        "execution_output": execution_output,
-    }
+    return_data['message'] = f"Function '{function_name}' called on contract at {contract_address} with args {args}."
+
+    return_data['data'] = execution_output
```
Comment (cristiam86) [inline rpc/server.py:407]:
```suggestion
    return_data['execution_output'] = execution_output
```
just to keep the format the same for now

File: genvm/server.py [hunk 2]
```diff
@@ -164,7 +161,64 @@ def get_icontract_schema(icontract: str) -> dict:
                     if hasattr(stmt.annotation, "id") and hasattr(stmt.target, "id"):
                         variables[stmt.target.id] = stmt.annotation.id
 
-    return {"class": class_name, "methods": methods, "variables": variables}
+    return {"class":class_name, "methods": methods, "variables": variables}
+
+
+@jsonrpc.method("deploy_contract")
```
Comment (cristiam86) [inline genvm/server.py:167]: why these methods appear as added if they are already in main just like that?

**Review by cristiam86 — CHANGES_REQUESTED** [review]
Please check the issues mentioned

**Comment by cristiam86** [comment]
Not needed anymore

**Pull request closed without merging**